packages/catalog-model/src/entity/ref.ts

```diff
@@ -196,3 +211,53 @@ export function serializeEntityRef(
 
   return `${kind ? `${kind}:` : ''}${namespace ? `${namespace}/` : ''}${name}`;
 }
+
+/**
+ * Matches an entity against either a string reference or a compound reference.
+ *
+ * The matching is case insensitive, and all of kind, namespace, and name must
+ * match (after applying the optional context to the ref).
+ *
+ * @param entity The entity to match
+ * @param ref A string or compound entity ref
+ * @param context An optional context of default kind and namespace, that apply
+ *                to the ref if given
+ * @returns True if matching, false otherwise
+ */
+export function matchEntityWithRef(
+  entity: Entity,
+  ref: EntityRef | EntityName,
+  context?: EntityRefContext,
+): boolean {
```

Comment:
Sometimes we also have to user cases that we have to refs that we want to compare, but I don't have a good example yet (and we already have an implementation for that somewhere). But not sure if it's worth to have a mega function for all cases 😆